modules/cloudsql-instance/variables.tf

```diff
@@ -172,6 +172,7 @@ variable "name" {
 variable "network" {
   description = "VPC self link where the instances will be deployed. Private Service Networking must be enabled and configured in this VPC."
   type        = string
+  default     = ""
```

Comment:
I don't we need need such a default. The other thing is - if I understand correctly, instance can use either PSA (then we need `network` + optional `allocated_ip_range`) or PSC (then we need `allowed_consumer_projects` and maybe `psc_enabled`). or the instance does not have private IP at all. I think we can improve the interface here by grouping all of those under one variable like `network_config`.

Something similar to what we have [here](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/modules/compute-vm/variables.tf#L315)  with either hourly, daily or weekly schedule.

Reply:
I'ts not clear from the docs but my understanding is that you still need the network (i.e. PSA) and you can optionally enable PSC on top of it:
> At least ipv4_enabled must be enabled or a private_network must be configured

If that's correct,n the this variable should remain mandatory

@luigi-bitonti can you confirm?

Reply:
If you enable PSC and private_network, you will receive an error. This is an open issue on the provider's source code that talks about it: : https://github.com/hashicorp/terraform-provider-google/issues/16257

Reply:
Hi @wiktorn, as you suggest I tried to introduce a new variable, network_config. Let me know if you think it's ok.

Reply:
Thank you for this